hail/src/main/scala/is/hail/io/CodecSpec.scala

```diff
@@ -17,19 +18,19 @@ trait CodecSpec extends Spec {
 }
 
 trait CodecSpec2 extends Spec {
-  def encodedType: Type
+  def encodedType: EType
+  def encodedVirtualType: Type
 
   type StagedEncoderF[T] = (Code[Region], Code[T], Code[OutputBuffer]) => Code[Unit]
   type StagedDecoderF[T] = (Code[Region], Code[InputBuffer]) => Code[T]
 
-  def buildEncoder(t: PType): (OutputStream) => Encoder = buildEncoder(t, t)
-  def buildEncoder(t: PType, requestedType: PType): (OutputStream) => Encoder
```

Comment:
how can you remove the requestedType ptype method?

Reply:
The type that's held by the `PackedCodecSpec2` is the requested type.